chia/full_node/full_node.py

```diff
@@ -2216,45 +2221,48 @@ async def respond_transaction(
         if not test and not (await self.synced()):
             return MempoolInclusionStatus.FAILED, Err.NO_TRANSACTIONS_WHILE_SYNCING
 
+        spend_bundle_id = bytes_to_SpendBundleID(spend_name)
```

Comment:
why not change the type of `spend_name` instead?

Reply:
> why not change the type of spend_name instead?

Because `respond_transaction` is near the interface to the one-the-wire protocol. More thought has to be put in before changing types up that call chain, so I figured this is a good place to cut the type propagation for now.